document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```
@@ -0,0 +1,347 @@
+          }
+        }
+        case AND: {
+          String childList =
+            Arrays.stream(filter.getChildFilters())
+              .map(this::parseQuery)
+              .filter(str -> !str.isEmpty())
+              .collect(Collectors.joining(" AND "));
+          if (!childList.isEmpty()) {
+            return childList;
+          } else {
+            return null;
+          }
+        }
+        default:
+          throw new UnsupportedOperationException(
+            String.format("Boolean operation:%s not supported", op));
+      }
+    } else {
+      Filter.Op op = filter.getOp();
+      Object value = filter.getValue();
+      String fieldName = filter.getFieldName();
+      String prefix = getFieldPrefix(fieldName);
+      StringBuilder filterString = new StringBuilder(prefix);
+      switch (op) {
+        case EQ:
+          filterString.append(" = ");
+          break;
+        case LIKE:
+          // Case insensitive regex search, Append % at beginning and end of value to do a regex search
+          filterString.append(" ILIKE ");
+          value = "%" + value + "%";
+          break;
+        case IN:
+          filterString.append(" IN ");
+          break;
+        case CONTAINS:
+          // TODO: Matches condition inside an array of documents
+          break;
+        case GT:
+          filterString.append(" > ");
+          break;
+        case LT:
+          filterString.append(" < ");
+          break;
+        case GTE:
+          filterString.append(" >= ");
+          break;
+        case LTE:
+          filterString.append(" <= ");
+          break;
+        case EXISTS:
+          // TODO: Checks if key exists
+          break;
+        case NOT_EXISTS:
+          // TODO: Checks if key does not exist
+          break;
+        case AND:
+        case NEQ:
+        case OR:
+          throw new UnsupportedOperationException("Only Equality predicate is supported");
+        default:
+          break;
```

In the default case, should we return the `null` string?

---

Yes, we can pass null on default case.

---

Actually, for default, we can also throw an unsupported exception, similar to what we do for composite filters.